rockcraft/models/project.py

```diff
@@ -133,7 +131,7 @@ def _validate_deprecated_build_base(cls, base_value: str | None) -> str | None:
 
     @staticmethod
     def _check_deprecated_base(base_value: str | None, field_name: str) -> str | None:
-        if base_value in DEPRECATED_COLON_BASES:
+        if base_value and base_value.startswith("ubuntu:"):
```

Comment:
The reason we special-cased `ubuntu:20.04` and `ubuntu:22.04` was to not break existing rockcraft projects when we moved to `ubuntu@`. For newer bases we want users to use `ubuntu@`, is there a specific reason to allow this now?

Reply:
For bases from 24.04 and onwards, it is just a consistency thing, because we don't want users to use `ubuntu:24.04`, right? (users coming from a docker/lxd background are more used to using colons).